relayer/src/event/rpc.rs

```diff
@@ -28,26 +21,29 @@ pub fn get_all_events(
 
             vals.push((height, NewBlock::new(height).into()));
         }
-
-        RpcEventData::Tx { .. } => {
-            let events = &result.events.ok_or("missing events")?;
-            let height_raw = events.get("tx.height").ok_or("tx.height")?[0]
-                .parse::<u64>()
-                .map_err(|e| e.to_string())?;
+        EventData::Tx { tx_result } => {
             let height = Height::new(
                 ChainId::chain_version(chain_id.to_string().as_str()),
-                height_raw,
+                tx_result.height as u64,
             );
-
-            let actions_and_indices = extract_helper(events)?;
-            for action in actions_and_indices {
-                if let Ok(event) = build_event(RawObject::new(
-                    height,
-                    action.0,
-                    action.1 as usize,
-                    events.clone(),
-                )) {
-                    vals.push((height, event));
+            for event in tx_result.result.events.iter() {
+                let abci_event = tendermint::abci::Event {
+                    type_str: event.event_type.to_string(),
+                    attributes: event
+                        .attributes
+                        .iter()
+                        .map(|rpc_tag| {
+                            serde_json::from_str::<AbciTag>(
+                                serde_json::to_string(rpc_tag).unwrap().as_str(),
```

Comment:
Can we write out the conversion manually/explicitly (perhaps via a `From` instance) instead of doing a roundtrip via JSON?

Reply:
i think we can remove the conversion altogether here since the types are now unified in tendermint-rs (`event` is now  `tendermint::abci::Event`).

Reply:
removed.